docs/Server.md

```diff
@@ -218,6 +218,35 @@ fastify.listen(3000)
 
 Internally Fastify uses the API of Node core HTTP server, so if you are using a custom server you must be sure to have the same API exposed. If not, you can enhance the server instance inside the `serverFactory` function before the `return` statement.<br/>
 
+<a name="schema-json-shorthand"></a>
+### `jsonShorthand`
+
+Internally, and by default, Fastify will automatically infer the root properties of JSON Schemas if it doesn't find valid root properties according to the JSON Schema spec. If you wish to implement your own schema validation compiler, for example: to parse schemas as JTD instead of JSON Schema, then you can explicitly set this option to `false` to make sure the schemas you receive are unmodified and are not being treated internally as JSON Schema.
+
+```js
+const AjvJTD = require('ajv/dist/jtd'/* only valid for AJV v7+ */)
+const ajv = new AjvJTD({
+  // This would let you throw at start for invalid JTD schema objects
+  allErrors: true
```

Comment:
Let's leave this out. It introduces a denial of service vulnerability.

Reply:
Would it be okay to replace it with `allErrors: process.env.node_env === 'development'`? If people are migrating, or have any JSON Schema code, it will be important to throw on invalid schemas, since otherwise all their JSON Schema schemas will just act as allow-all schemas resulting in no input validation.

Reply:
I think that is fair.